extension/wallet/src/lib_wallet/extract_loan.rs

```diff
@@ -0,0 +1,79 @@
+use crate::lib_wallet::{LoanDetails, Wallet};
+use baru::loan::{Borrower0, Borrower1, LoanResponse};
+use elements::{secp256k1_zkp::SECP256K1, AssetId};
+
+pub async fn extract_loan<W: Wallet + Sync>(
+    wallet: &W,
+    loan_response: LoanResponse,
+    btc_asset_id: AssetId,
+    usdt_asset_id: AssetId,
+    borrower0: Borrower0,
+) -> Result<(Borrower1, LoanDetails), Error> {
```

Comment:
Let's try and make as many functions of this _data-driven_ instead of "dependency"-driven, i.e.:

Take in a list of utxos as a parameter and define your functionality based on that.

A good rule of thumb is: Only depend on functionality through a type parameter if it is conditional on something else. In this case, you can just pass in a list of UTXOs. A value-driven API is always more flexible which is what you want in a library.